lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -3139,6 +3139,47 @@ class DecomposeAtenMishOp : public OpRewritePattern<AtenMishOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.mish` op into `aten.tanh` and `aten.softplus` ops.
+// Mish(x) = x * Tanh(Softplus(x))
+class DecomposeAtenXlogyTensorOp : public OpRewritePattern<AtenXlogyTensorOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenXlogyTensorOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto context = op.getContext();
+    auto resultType = op.getType().cast<BaseTensorType>();
+    Value constZero = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(0));
+
+    auto eqType = ValueTensorType::get(
+        context, op.getType().cast<BaseTensorType>().getSizes(),
+        IntegerType::get(context, 1));
```

Comment:
There is no guarantee that the types will have value semantics at this stage. It is better to create the type from the existing types:

```c++
auto resultType = op.getType().cast<BaseTensorType>();
auto eqType = resultType.getWithSizesAndDtype(
    resultType.getOptionalSizes(), IntegerType::get(context, 1));
```